pallets/circuit-portal/src/xbridges/mod.rs

```diff
@@ -62,9 +65,45 @@ pub fn get_roots_from_bridge<T: pallet_multi_finality_verifier::Config<I>, I: 's
             gateway_block_hash,
         )
         .ok_or(Error::<T>::StepConfirmationBlockUnrecognised)?;
+
     Ok((extrinsics_root, storage_root))
 }
 
+pub fn read_cmp_latest_height_from_bridge<
```

Comment:
Looks good. How would we go about exposing this via RPC? The ranger will also need access to this, to enable the `catch_up` logic we spoke about.

Reply:
Added to `read latest gateway height` RPC Runtime and RPC. Also cleaned up unused execute composable step via Portal